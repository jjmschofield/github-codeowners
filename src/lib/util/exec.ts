import { exec as realExec } from 'child_process';
import { promisify } from 'util';

export const exec = promisify(realExec);
