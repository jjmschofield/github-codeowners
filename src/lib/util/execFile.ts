import { execFile as realExecFile } from 'child_process';
import { promisify } from 'util';

export const execFile = promisify(realExecFile);
