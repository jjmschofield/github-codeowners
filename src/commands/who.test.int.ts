import { v4 as uuidv4 } from 'uuid';
import fixtures from './__fixtures__/default';
import { generateProject } from './__fixtures__/project-builder.test.helper';

import util from 'util';

const exec = util.promisify(require('child_process').exec);

describe('who', () => {
  const testId = uuidv4();

  let testDir = 'not set';

  beforeAll(async () => {
    testDir = await generateProject(testId, fixtures);
    // tslint:disable-next-line:no-console
    console.log(`test scratch dir: ${testDir}`);
  });

  const runCli = async (args: string) => {
    return exec(`node  ../../../dist/cli.js ${args}`, { cwd: testDir });
  };

  it('should list ownership for all files', async () => {
    const { stdout, stderr } = await runCli('who default-wildcard-owners.md');
    expect(stdout).toMatchSnapshot('stdout');
    expect(stderr).toMatchSnapshot('stderr');
  });
});
