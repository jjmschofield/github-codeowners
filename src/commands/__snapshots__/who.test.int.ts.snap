// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`who should list ownership for all files: stderr 1`] = `""`;

exports[`who should list ownership for all files: stdout 1`] = `
"default-wildcard-owners.md	@global-owner1	@global-owner2
"
`;
